src/dict_support.jl

```diff
@@ -1,12 +0,0 @@
-# support functions
-
-# _tablesz and hashindex are defined in Base, but are not exported,
-# so they are redefined here.
-_tablesz(x::Integer) = x < 16 ? 16 : one(x)<<((sizeof(x)<<3)-leading_zeros(x-1))
-hashindex(key, sz) = (reinterpret(Int,(hash(key))) & (sz-1)) + 1
-
-function not_iterator_of_pairs(kv)
```

Comment:
I am not please at depending on importing this from 
*OrderedCollections*,
because out other dicts need it, 
and ideally we'ld be able to cut them out into their own packages, 
which would also not depend on `OrderedCollections`

I see two alternative options.
1. just duplicate the code. I don't mind this, while it is not DRY, it  does let us evolve the remainind dicts independently of **OrderedCollections**
2. make a micro package **DataStructuresUtil.jl** or some such, and have both import this.

Reply:
Interestingly, I think OrderedCollections doesn't even use that function. So I can put it back here and delete it from there.